src/components/widgets-common/chart-threshold/chart-threshold.component.coffee

```diff
@@ -148,19 +155,25 @@ module.component('chartThreshold', {
 
     shrinkChart = ->
       return unless ctrl.chart
-      ctrl.chart.setSize(null, ctrl.chart.chartHeight - ctrl.chartShrinkSize, false)
-      ctrl.chart.container.parentElement.style.height = "#{ctrl.chart.chartHeight}px"
+      ctrl.chart.hc.setSize(null, ctrl.chart.hc.chartHeight - ctrl.chartShrinkSize, false)
+      ctrl.chart.hc.container.parentElement.style.height = "#{ctrl.chart.hc.chartHeight}px"
 
     growChart = ->
       return unless ctrl.chart
-      ctrl.chart.setSize(null, ctrl.chart.chartHeight + ctrl.chartShrinkSize, false)
-      ctrl.chart.container.parentElement.style.height = "#{ctrl.chart.chartHeight}px"
+      ctrl.chart.hc.setSize(null, ctrl.chart.hc.chartHeight + ctrl.chartShrinkSize, false)
+      ctrl.chart.hc.container.parentElement.style.height = "#{ctrl.chart.hc.chartHeight}px"
 
     # Disable threshold when selected time period is strictly in the past
     validateHistParameters = ->
       widgetHistParams = ctrl.widget.metadata && ctrl.widget.metadata.hist_parameters
       ctrl.disabled = widgetHistParams? && moment(widgetHistParams.to) <= moment.utc().startOf('day')
-      return
+      return !ctrl.disabled
+
+    # Validate and build threshold data from widget kpi templates
+    buildThresholdsFromKpis = ->
+      targets = ctrl.widget.kpis? && ctrl.widget.kpis[0] && ctrl.widget.kpis[0].targets
+      return [] unless ImpacKpisSvc.validateKpiTargets(targets)
+      [{ kpiId: ctrl.widget.kpis[0].id, value: targets.threshold[0].min, name: 'Alert Threshold', color: ctrl.thresholdColor, onClickEvent: onThresholdClick }]
```

Comment:
@cesar-tonnoir I have left the threshold name static for now as I don't think it makes sense for the widget controller to set this (how would that if / when this is extended to iterate over multiple widget kpis?), and the UI/UX requirements for the name displayed have not been defined as of yet.

Reply:
ok